kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemDataset.java

```diff
@@ -562,4 +572,35 @@ public boolean isEmpty() {
     }
   }
 
+  @Override
+  public void signalReady() {
+    unbounded.signalReady();
+  }
+
+  @Override
+  public boolean isReady() {
+    return unbounded.isReady();
+  }
+
+  public void signalReady(Constraints viewConstraints) {
+    Preconditions.checkNotNull(viewConstraints, "Constraints cannot be null");
+
+    Path signalsPath = new Path(directory, SIGNALS_DIRECTORY_NAME);
+
+    SignalManager signalManager = SignalManager.create(fileSystem.getConf(), signalsPath);
```

Comment:
The change commented [here](https://github.com/kite-sdk/kite/pull/348#discussion_r29079115) removed this awkwardness. The dataset isReady/signalReady methods still call into the unbounded view, but the call back up into the dataset is removed.